app/stores/AppStore.js

```diff
@@ -32,23 +34,30 @@ export default class AppStore extends Store {
 
   @computed get currentLocale(): string {
     if (!this.isCurrentLocaleSet) {
+      this.getProfileLocaleRequest.execute();
       return 'en-US';
     }
-    return this.profileLocaleRequest.result;
+    return this.getProfileLocaleRequest.result || this.setProfileLocaleRequest.result;
+  }
+
+  @computed get hasLoadedCurrentLocale(): bool {
+    return this.getProfileLocaleRequest.wasExecuted;
   }
 
   @computed get isCurrentLocaleSet(): bool {
-    return this.profileLocaleRequest.wasExecuted;
+    if (this.getProfileLocaleRequest.result == null) return false;
+    return (this.getProfileLocaleRequest.wasExecuted && this.getProfileLocaleRequest.result) ||
+           (this.setProfileLocaleRequest.wasExecuted && this.setProfileLocaleRequest.result);
```

Comment:
@tomothespian and I discussed this yesterday and were thinking to do it the way you suggested BUT we wanted to avoid additional call to getProfileLocaleRequest  since setProfileLocaleRequest is returning saved locale already ;)
But I agree that it will be much nicer the way you suggested!